src/gwt/src/org/rstudio/studio/client/workbench/views/vcs/BranchToolbarButton.java

```diff
@@ -176,14 +184,29 @@ public HandlerRegistration addValueChangeHandler(ValueChangeHandler<String> hand
    @Override
    public void onVcsRefresh(VcsRefreshEvent event)
    {
-      JsArrayString branches = pVcsState_.get().getBranchInfo().getBranches();
-      onRefresh(branches);
+      JsVectorString branches = pVcsState_.get().getBranchInfo().getBranches().cast();
+      if (branches.length() != branches_.length())
+      {
+         branches_ = branches.slice().cast();
+         menuRebuildRequired_ = true;
```

Comment:
Nit: should we bail out here? (seems like once we've determined a rebuild is required there's no need to compare each branch individually)

Reply:
We should -- thanks for catching that!